packages/oats-runtime/src/server.ts

```diff
@@ -131,11 +131,8 @@ function voidify(value: object | undefined | null) {
 function cleanHeaders<H>(mode: Mode, maker: Maker<any, H>, headers: object) {
   // note: we now expect that on client side headers are type conforming so do not need to lowercase those
   // this is slightly breaking change is somebody has subverted the type checker
-  const normalized = voidify(mode === 'server' ? lowercaseObject(headers) : headers);
-  const acceptsNull = maker(null);
```

Comment:
I don't understand this. We return null no matter what the headers are if null is accepted? That seems just wrong

Reply:
> if null is accepted

null is only accepted if there are no headers. Bc otherwise, the schema will be of type `object`, not `void`. Basically we're relying on an implementation detail here (i.e. we don't have a schema that accepts both null and objects).